pynopegl-utils/pynopegl_utils/qml/livectls.py

```diff
@@ -20,45 +20,64 @@
 # under the License.
 #
 
+from dataclasses import dataclass
 from typing import Any, Dict, List
 
 from PySide6.QtGui import QColor
 
 import pynopegl as ngl
 
-_NODE_TYPES_MODEL_MAP = dict(
-    UniformColor="color",
-    UniformBool="bool",
-    UniformVec2="vector",
-    UniformVec3="vector",
-    UniformVec4="vector",
-    UniformMat4="vector",
-    Text="text",
-)
 
-_NODE_TYPES_CMP_MAP = dict(
-    UniformVec2=2,
-    UniformVec3=3,
-    UniformVec4=4,
-    UniformMat4=4 * 4,
+@dataclass
+class _ControlSpec:
+    control_type: str
+    base_type: str
+    components: int
+
+
+_NODE_TYPES_SPEC_MAP = dict(
+    Text=_ControlSpec("text", "str", 1),
+    UniformBool=_ControlSpec("bool", "bool", 1),
+    UniformColor=_ControlSpec("color", "float", 1),
```

Comment:
1 component?

Reply:
I went with one component as it rely on solely one widget. I can put 3 if you prefer.